### PR TITLE
mds: convert const char ptr to string_view

### PR DESCRIPTION
**Pull request by batrick**

### CODE REVIEW

File: src/mds/MDSTable.h [hunk 1]
```diff
@@ -27,7 +27,7 @@ class MDSTable {
 public:
   MDSRank *mds;
 protected:
-  const char *table_name;
+  std::string table_name;
```
Comment (tchaikov) [inline src/mds/MDSTable.h]: why not just use string_view? as we assume the content of string_view is always null terminated.

File: src/common/TrackedOp.h [hunk 2]
```diff
@@ -234,35 +234,26 @@ class TrackedOp : public boost::intrusive::list_base_hook<> {
 
   struct Event {
     utime_t stamp;
-    string str;
-    const char *cstr = nullptr;
+    std::string str;
```
Comment (tchaikov) [inline src/common/TrackedOp.h]: this forces us to create a `std::string` from `string_view`. IMHO, it practically defeats the purpose of using `string_view`. i'd use `std::variant<>` for holding the `string_view` or `std::string`.

Comment (batrick) [reply src/common/TrackedOp.h]: The `std::string_view` may not refer to static memory? In several places, an Event is created with `s` referring to a temporary string.

File: src/mds/MDSTable.h (hunk 1 above)
Comment (batrick) [reply src/mds/MDSTable.h]: `MDSTable` is only statically instantiated a few times in the MDS so the cost here is low.

I'm also concerned about having a `const char *` or `std::string_view` which refers to memory the class does not control/reference.

File: src/common/TrackedOp.h (hunk 2 above)
Comment (tchaikov) [reply src/common/TrackedOp.h]: i think we can differentiate that case at the caller site, can't we? and instead move the temporary string or copy it to the one in `Event`.

Comment (batrick) [reply src/common/TrackedOp.h]: I agree with you. See my comment here: https://github.com/ceph/ceph/pull/28849#discussion_r301836379

**Pull request merged**